### PR TITLE
2017 Summary Post

### PR DESCRIPTION
**Pull request by hennevogel**

### CODE REVIEW

**Comment by hennevogel** [comment]
It would be cool if one of you could add a numbers section with github statistics about 2017. Something like the [monthly pulse](https://github.com/openSUSE/open-build-service/pulse/monthly) but for the whole year. 

screenshot from 2017-12-21 18-25-05

I don't think github generates this but this should be doable with a tool like [gitstats](http://gitstats.sourceforge.net/) (please make sure to personalize this to the lowest extent)

**Comment by hennevogel** [comment]
Found my old `gitstats` installation, here are the numbers.

**37 authors** have pushed **3219 commits**. On average that are **17.41 authors** pushing **268.25 commits** per month. On [master](https://github.com/openSUSE/open-build-service/compare/3464c6674f513c809f15ba2466087afaf9da42a6...master), **2335 files** have changed and there have been **278759 additions** and **114210 deletions**. **1060 Pull Requests have [been merged](https://github.com/openSUSE/open-build-service/pulls?utf8=✓&q=is%3Apr+merged%3A>%3D2017-01-01)**, **210 [issues](https://github.com/openSUSE/open-build-service/issues?utf8=%E2%9C%93&q=is%3Aissue+created%3A%3E%3D2017-01-01+) opened** and issues **292 closed**. All in all this makes 2017 the busiest year in the history of the Open Build Service. $UCCE$$!

## Commits by Year
commits_by_year

**Comment by hennevogel** [comment]
Did everything myself now. Please review!

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+title: Frontend Team's 2017
+author: Henne Vogelsang
+category: development
+---
+The year is coming to an end, all the 🎁 lie under the 🎄 and the 🍾 is already cold for 🎆. Being the good agile folks that we are, we think this is the perfect time to reflect on 2017. On all the code, the bugs and fixes, features and refactorings, the tunes and adjustments we have introduced to our code base, process and tool belt. Not only to trace the steps of our journey, but also to get ideas how to become more effective in 2018.
+
+So let's look at some of the highlights that have happened this year. Bear with us, this is a very long post, but one worth reading, promised!
+
+## January
+The year started off with some changes to the <a href="/team">team</a>, Evan joined mid-December 2016 so beginning of this year he was still shiny and had this new developer smell we all like so much. Code wise the year kicked off with a Ruby 2.4 port, the introduction of property tests into our suite and a new API route to get the GPG key and SSL certificate for a project.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "so **at the** beginning of this year"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+January is also around the time we start to invite guests to our review meeting to expose more people to what we achieve.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "**started** to invite"
"review **meetings**"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## February
+February is traditionally the time Free Software developers gather in Brussels for FOSDEM. So a couple of us went there and Christian held a talk about [Continuous Integration with the Open Build Service](https://archive.fosdem.org/2017/schedule/event/continuous_integration_with_obs/).
+
+After the event, we used the opportunity that people already where on the move to hold our first "*official*" get together as a team in the SUSE headquarters. We kicked off the **Grand OBS Development Boot-camp** where we met many of our stakeholders (distribution packagers & maintainers, OBS admins and customers etc.) to understand how they utilize what we produce. We also talked about our mid/long term plans and explored new technologies together in one room. For us as distributed team, that works from all over Europe, being together is always very refreshing.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "already **were** on the move"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+May is also the month where we first started to publish [sprint reports](http://openbuildservice.org/2017/05/05/frontend-sprint-report-1/) to make it even easier to follow the OBS frontend development. At the end of the month most of us gathered in Nürnberg once again, to organize, attend and speak at the openSUSE Conference. We contributed 8(!) talks & workshops about various topics to the oSC17 schedule.
+
+## June
+Because someone invented the beer garden and other summer goodies like vacation, June was a bit more quiet. 
+Nevertheless we introduced a page to the UI where you can look up the reason why your package has been built and some of us started to hack away at a big refactoring of `BsRequest.collection` and a first stab at extending our `Event` subsystem.
+
+## July
+The next month we where picking up pace again. An unexpected downtime of the backend left us with 2+ million delayed jobs our queues. This amount exposed many issues with this sub-system so a bunch of us really dived into this, head first.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: I would get rid of "really"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+July was also the month where we started to make use of the brilliant [depfu](https://depfu.com/) service (hey Florian 💚), and the month we wrote our [first post-mortem report](http://openbuildservice.org/2017/07/04/post-mortem-1/) for failed deployments.
+
+## August
+In August we start to use [codeclimate](https://codeclimate.com/github/openSUSE/open-build-service) and [git-cop](https://github.com/bkuhlmann/git-cop) in pull requests, shipped another bugfix release (2.8.3) and started to use [build stages](https://blog.travis-ci.com/2017-05-11-introducing-build-stages) for our CI cycle on Travis.
+
+Moises introduced a brand new backend abstraction layer (`Backend::Api`), we based all our delayed jobs on `ActiveJob` and fixed a lot of issues with this sub-system while at it.
+
+## September
+The next month we introduced the concept of a **demolition squad**, a 2 people team that is responsible for all the repetitive tasks (issue/exception triage, package/gem updates, deployments and so on) for a week. This spread this knowledge fast, by end of September we deployed on 4 out of 5 days a week, so we practically switched to continuous deployment! 🤡
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "2 **person** team"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+We collaborated with the SUSE documentation team to [refresh all our OBS books/guides](http://openbuildservice.org/2017/09/22/two-new-guides-for-builders/), dockerized our development environment and a bunch of us went to Euruko in Budapest. Björn celebrated a full decade at SUSE, Alex started his apprenticeship in our team and sadly it's also the month we see another departure from the team, as we all have to wish Jan farewell because he leaves SUSE.
+
+But September also the month we finally launch the *Studio Express* workflow to the public that we have worked on for so long!
+
+## October
+In October we finish off the implementation of a RabbitMQ based message bus for the OBS, update the hashing algorithm of the passwords in our database and enable notifications for projects in your watchlist. Eduardo and Björn hold a well received [workshop at the Open Source Summit Europe](https://osseu17.sched.com/event/ByRq/workshop-continuous-integration-with-the-open-build-service-eduardo-navarro-bjorn-geuken-suse-linux-gmbh) and Ana visits openSUSE.asia.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "**finished** off"

Comment (evanrolfe) [reply _posts/development/2017-12-20-summary-of-2017.md]: "**updated** the hashing algorithm"

Comment (evanrolfe) [reply _posts/development/2017-12-20-summary-of-2017.md]: "**enabled** notifications"

Comment (evanrolfe) [reply _posts/development/2017-12-20-summary-of-2017.md]: "Ana **visited**"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+## November
+Last month the [shutdown of SUSE Studio](http://blog.susestudio.com/2017/11/shut-down-of-suse-studio-at.html) is announced, another round of innovative coding happens at SUSEs Hackweek 16 and we gather for our second workshop at the SUSE headquarter in Nürnberg. There we mostly talk about how we can replace the OBS frontend technology, which is a bit dated by now, and how to extend the *Studio Express* workflow.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "**was** announced"
"**happened** at"
"**gathered** at"
"headquarters"
"talked"
"which was a bit.."

This all happened in the past so needs to be past tense! :)

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+In the code-base we see the first contributions from Alex, lots of *Studio Express* workflow fixes and the usual round of refactorings and bugfixes.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "we **saw**"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+## December
+This month we continued with improvements to our SCRUM process: As we spend 30% of our time in the sprint on development tasks (refactoring, extending the test suite, write architectural documentation etc.) we introduced the concept of a *Development Owner* additionally to the *Product Owner*. The **DO** is responsible for grooming/planing development tasks.
```
Comment (evanrolfe) [inline _posts/development/2017-12-20-summary-of-2017.md]: "spent"
"**writing** architectural"

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+We also integrated the backend tests into our CI cycle on Travis, refactored the build results and participated in SUSEs Hack Week 15, where everyone was busy exploring and learning new things.
+
+## March
+In March we introduced pundit for authorization into our attribute system, some new [service classes](https://blog.engineyard.com/2014/keeping-your-rails-controllers-dry-with-services) for the `Project` model and [foreman](https://rubygems.org/gems/foreman) to manage all processes that need to run for our application.
+
+Henne had his 17(**!**) year anniversary with SUSE, we helped run the [SUSE Students Day](https://github.com/ChrisBr/mentoring/wiki/SUSE-Students-Day) and also helped to make room for new servers by moving the complete SUSE Studio rack to another floor with less than 6 hours downtime. Yay!
```
Comment (ChrisBr) [inline _posts/development/2017-12-20-summary-of-2017.md]: @evanrolfe can you please replace this link with 
https://www.suse.com/c/students-day-suse-talents-sight/

The GitHub wiki page was just the draft ...

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+title: Frontend Team's 2017
+author: Henne Vogelsang
+category: development
+---
+The year is coming to an end, all the 🎁 lie under the 🎄 and the 🍾 is already cold for 🎆. Being the good agile folks that we are, we think this is the perfect time to reflect on 2017. On all the code, the bugs and fixes, features and refactorings, the tunes and adjustments we have introduced to our code base, process and tool belt. Not only to trace the steps of our journey, but also to get ideas how to become more effective in 2018.
+
+So let's look at some of the highlights that have happened this year. Bear with us, this is a very long post, but one worth reading, promised!
+
+## January
+The year started off with some changes to the <a href="/team">team</a>, Evan joined mid-December 2016 so at beginning of this year he was still shiny and had this new developer smell we all like so much. Code wise the year kicked off with a Ruby 2.4 port, the introduction of property tests into our suite and a new API route to get the GPG key and SSL certificate for a project.
```
Comment (ChrisBr) [inline _posts/development/2017-12-20-summary-of-2017.md]: [nitpick] Evan and David!

IIRC David started at around the same time ...

File: _posts/development/2017-12-20-summary-of-2017.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+We also shipped another bugfix release (2.8.2), introduced the *Studio Express* repository/package list editor and made it possible to subscribe to an RSS feed for your personal notifications. To address the rising demand for LDAP support we also looked deeper into this code and respective tests.
+
+July was also the month where we started to make use of the brilliant [depfu](https://depfu.com/) service (hey Florian 💚), and the month we wrote our [first post-mortem report](http://openbuildservice.org/2017/07/04/post-mortem-1/) for failed deployments.
+
+## August
+In August we start to use [codeclimate](https://codeclimate.com/github/openSUSE/open-build-service) and [git-cop](https://github.com/bkuhlmann/git-cop) in pull requests, shipped another bugfix release (2.8.3) and started to use [build stages](https://blog.travis-ci.com/2017-05-11-introducing-build-stages) for our CI cycle on Travis.
+
+Moises introduced a brand new backend abstraction layer (`Backend::Api`), we based all our delayed jobs on `ActiveJob` and fixed a lot of issues with this sub-system while at it.
+
+## September
```
Comment (ChrisBr) [inline _posts/development/2017-12-20-summary-of-2017.md]: In September we also had the booth at SUSE Con, maybe we should mention that as well

Comment (evanrolfe) [reply _posts/development/2017-12-20-summary-of-2017.md]: Can you write a sentence about it? I dont know anything about it so am not sure what to write..

Comment (ChrisBr) [reply _posts/development/2017-12-20-summary-of-2017.md]: I would write as last sentence of the second paragraph:

> Beside all that, we were also involved managing the Open Build Service booth at the annual SUSECon in Prague.

**Pull request merged**